apps/spruce/src/components/styles/Link.tsx

```diff
@@ -19,14 +19,31 @@ export const StyledLink = (props) => (
 
 // @ts-expect-error: FIXME. This comment was added by an automated script.
 export const StyledRouterLink = (props) => <StyledLink as={Link} {...props} />;
-
-export const ShortenedRouterLink = styled(StyledRouterLink)<{
-  width?: number;
-}>`
+interface ShortenedRouterLinkProps {
+  baseWidth?: number;
+  responsiveBreakpoint?: number;
+}
+/**
+ * ShortenedRouterLink is a styled component that truncates the text of a link and adds an ellipsis if it overflows.
+ * @param props The props for the ShortenedRouterLink component.
+ * @param props.baseWidth The base width of the link.
+ * @param props.responsiveBreakpoint The breakpoint at which the link should set it's width based on screen width.
+ * @returns A styled link that truncates the text and adds an ellipsis if it overflows.
+ */
+export const ShortenedRouterLink = styled(
+  StyledRouterLink,
+)<ShortenedRouterLinkProps>`
   span {
     display: inline-block;
     vertical-align: bottom;
-    max-width: ${({ width }) => `${width ?? 200}px`};
+    ${({ baseWidth, responsiveBreakpoint }) =>
+      responsiveBreakpoint
+        ? `@media (max-width: ${responsiveBreakpoint}px) { max-width: ${baseWidth}px; }`
+        : null};
+    max-width: ${({ baseWidth, responsiveBreakpoint }) =>
+      responsiveBreakpoint
+        ? `calc(100vw - ${responsiveBreakpoint - (baseWidth ?? 200)}px)`
+        : `${baseWidth ?? 200}px`};
```

Comment:
Sorry for the slow review, but I was wondering why would this approach be advantageous as opposed to just giving it a percentage. `max-width: 50%` or something like that would accomplish the same effect of allowing the link to scale in width in relation to the screen size while avoiding this breakpoint based logic.

Reply:
The current solution tries to set the max-width as close as it can the length of the red line. This is fr efficient use space and preventing the truncated text from taking up an entire line of it's own. The black line is the breakpoint value. The value is configurable because of differing page layouts.
![Screenshot 2024-08-22 at 10 57 38 AM](https://github.com/user-attachments/assets/965afb3d-c8f5-4dd3-ae7e-754519a4f564)

Here is a video of the current solution: 

https://github.com/user-attachments/assets/09ee5792-d967-42db-8904-28d44a0ff1d1

Here is a video of setting max-width to 50vw:

https://github.com/user-attachments/assets/9011daeb-40fc-4b3d-9408-5c5a7c039b78

Here is a video of setting max-width to 75vh:

https://github.com/user-attachments/assets/9692b9b4-4ab0-4a9f-a117-29655aa8cc25

I think the breakpoint solution looks best and allows for the most configuration at the cost of some code complexity.